docs/design/task-DAG-scheduling.md

```diff
@@ -0,0 +1,85 @@
+# Task-level DAG scheduling policy
+
+@[hwdef](https://github.com/hwdef); July 19th, 2021
+
+## Introduction
+
+This feature provides the ability to customize the order in which tasks are launched.
+
+## Scenarios
+
+* mpi job. the master needs to wait for the worker to start before starting, If the master is already started, but the worker is not yet started, the master will restart, which will add unnecessary waste of resources ,in this case, mpiworker needs to be in the running state, and then create the pod for mpimaster..
+
+* Suppose there are two tasks and task 2 needs to use the calculation results of task 1,in this case, task1 needs to be in the complete state, and then create the pod for task2.
+
+* some ETL applications
+
+## Requirement
+
+Each task can specify which one or more tasks it depends on, and it can specify the time point or interface for triggering.
+
+Examples of time points:
+
+* running
+* has been running for 10s
+* complete
+
+Examples of interfacets:
+
+* Check if a file named raw exists in the /data directory.
+* Check if the specified port is open. such as 2222.
+* Check if the http GET request to the specified path was successfully accessed.such as /health.
+
+## Design
+
+1. Add a field that is an array named `startPolicy` in task spec
+
+```go
+type TaskSpec struct {
+    ...
+	StartPolicy []StartPolicySpec
+}
+```
+2. It contains three fields, `dependOn`, `trigger`, `probe`
+
+```go
+type StartPolicySpec struct {
+	DependOn string
+	Trigger  TriggerSpec
+	Probe    *v1.Probe
+}
+```
+2.1 `dependOn` is a string indicating the name of the task that the current task depends on
+
+2.2 `trigger` indicates the point in time when the trigger, it has two fields, `status` and `time`, `status` indicates the status of the task, `time` indicates the duration of time since the state was reached,it can be empty, indicating 0 seconds.
+
+```go
+type TriggerSpec struct {
+	Status string
+	Time   time.Duration
+}
+```
+
+2.3 `probe` is used to detect the status of the task in dependOn. This field can be used with the `probe` struct in kubernetes
```

Comment:
I don’t quite understand what the field `probe ` is used for, can you elaborate more?

Reply:
I think probe applies to both cases in lines 29 and 30 of the documentation

Reply:
So do you mean the probe configured here should be applied to each containers in pod template?

Reply:
I think for both HTTP and TCP, there is no need to consider the issue of multiple containers. But `exec` method should be selected containers.

Reply:
I can think of two ways:
* Add a field to specify the container name
* dependOn.taskname in the format of `taskName/containerName`

Reply:
IMO, we should only care about taskName in task spec, so this field is not so necessary

Reply:
If don't add fields, then how to specify the container? Do you have any suggestions?

Reply:
> If don't add fields, then how to specify the container? Do you have any suggestions?

To simplify the implementation, i think we can only consider one container per pod temporarily for most users take it into practice.